src/lava/utils/float2fixed.py

```diff
@@ -1,26 +1,719 @@
+
+        # Initialize scale domain dictionary.
+        scale_domains = {}
+        for scale_domain, p_id in self.scale_domains:
+            # Only create entry if scale_domain has now entry in dict yet.
+            if scale_domain not in scale_domains.keys():
+                scale_domains[scale_domain] = {}
+            scale_domains[scale_domain][p_id] = {}
+
+        for p_id, conv_data_p_id in self.conv_data.items():
+            for var, conv_data in conv_data_p_id.items():
+                # Create copy of conv_data.
+                conv_data_copy = conv_data.copy()
+
+                # Get scale domain and remove it from dictionary.
+                scale_domain = conv_data_copy.pop('scale_domain')
+                # Remove other information not needed anymore.
+                conv_data_copy.pop('constant')
+                conv_data_copy.pop('exp_var')
+
+                if conv_data_copy['num_bits_exp'] is None:
+                    conv_data_copy['num_bits_exp'] = 0
+                scale_domains[scale_domain][p_id][var] = conv_data_copy
+
+        return scale_domains
+
+    def _find_scaling_factors(self) -> dict:
+        """Determine the optimal scaling function for the float- to fixed-point
+        mapping. The function must be consistent between Variables of one scale
+        domain of be chosen such that for every Variable the numerical
+        fixed-point representation can be represented with the given
+        constraints. The optimal scaling function will be stored in a
+        dictionary with keys scale domains.
+
+        Returns
+        -------
+        scaling_funct_dict: dict
+            Dictionary containing the optimal scaling function for each scale
+            domain.
+        """
+        # Initialize scaling function dictionary.
+        scaling_factors = {}
+        slopes = []
+
+        for scale_domain, scale_data_domain in self.scale_domains.items():
+            scaling_factors[scale_domain] = {}
+            for p_id, scale_data_p_id in scale_data_domain.items():
+                for var, scale_data in scale_data_p_id.items():
+
+                    bits = scale_data['precision_bits']
+                    impl_shift = scale_data['implicit_shift']
+                    num_bits_exp = scale_data['num_bits_exp']
+
+                    # Get range representable given bit constraints and
+                    # signedness.
+                    if scale_data['signedness'] == 's':
+                        rep_range = np.array([-1 * 2 ** (bits - 1),
+                                              2 ** (bits - 1) - 1])
+                    elif scale_data['signedness'] == 'u':
+                        rep_range = np.array([0, 2 ** bits - 1])
+
+                    # Apply implicit shift.
+                    rep_range = np.left_shift(rep_range, impl_shift)
+
+                    # Apply maximal shift due to exponent.
+                    rep_range = np.left_shift(rep_range, 2 ** num_bits_exp - 1)
```

Comment:
Given that you're a mathematician, a general question:
For quite a while, I was confused in our team that _exp typically refers to powers of two. Intuitively, it refers to an exponential function. Can you think of any other way how to abbreviate this?

Reply:
In principal I agree.  
But I don't really know a better name, something like "exponent to basis 2" maybe, but this does not admit a nice abbreviation.

But if you have a better idea than `num_bits_exp`  I readily change it.